src/platform/datapath_raw_socket.c

```diff
@@ -392,7 +407,25 @@ CxPlatResolveRoute(
         "Starting to look up neighbor on Path[%hhu] with status %u",
         PathId,
         Status);
-    if (Status != ERROR_SUCCESS || IpnetRow.State <= NlnsIncomplete) {
+    if ((Status != ERROR_SUCCESS || IpnetRow.State <= NlnsIncomplete) ||
+        (State == RouteSuspected &&
+         memcmp(
+             Route->NextHopLinkLayerAddress,
+             IpnetRow.PhysicalAddress,
+             sizeof(Route->NextHopLinkLayerAddress)) == 0)) {
+        //
+        // For route refresh, if the next hop has not changed, we will
+        // force neighbor discovery.
+        //
+        Status = ERROR_IO_PENDING;
+        CXPLAT_STATIC_ASSERT(
+            SUCCESS_HRESULT_FROM_WIN32(ERROR_IO_PENDING) == QUIC_STATUS_PENDING,
+            "SUCCESS_HRESULT_FROM_WIN32(ERROR_IO_PENDING) must be QUIC_STATUS_PENDING");
+    } else {
+        CxPlatResolveRouteComplete(Context, Route, IpnetRow.PhysicalAddress, PathId);
+    }
+
+    if (Status == ERROR_IO_PENDING) {
```

Comment:
Again an unidiomatic use of "pending". We can just put this block of code above where status is currently set to pending.